xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-schemes/xwiki-platform-url-scheme-standard/src/main/java/org/xwiki/url/internal/standard/StandardStringEntityReferenceResolver.java

```diff
@@ -0,0 +1,101 @@
+ */
+package org.xwiki.url.internal.standard;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.internal.reference.DefaultStringEntityReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.resource.CreateResourceReferenceException;
+import org.xwiki.resource.CreateResourceTypeException;
+import org.xwiki.resource.ResourceReference;
+import org.xwiki.resource.ResourceReferenceResolver;
+import org.xwiki.resource.ResourceType;
+import org.xwiki.resource.ResourceTypeResolver;
+import org.xwiki.resource.UnsupportedResourceReferenceException;
+import org.xwiki.resource.entity.EntityResourceReference;
+import org.xwiki.stability.Unstable;
+import org.xwiki.url.ExtendedURL;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Attempts to extract the {@link EntityReference} from a {@link String} URL, using the
+ * {@link StandardExtendedURLResourceTypeResolver}.
+ * 
+ * @version $Id$
+ * @since 14.0
+ */
+@Component
+@Named("resource/standard")
+@Singleton
+@Unstable
+public class StandardStringEntityReferenceResolver extends DefaultStringEntityReferenceResolver
+{
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Inject
+    @Named("standard")
+    private ResourceReferenceResolver<ExtendedURL> resourceResolver;
+
+    @Inject
+    @Named("standard")
+    private ResourceTypeResolver<ExtendedURL> typeResolver;
+
+    @Inject
+    private EntityReferenceResolver<String> defaultStringResolver;
+
+    @Override
+    public EntityReference resolve(String urlStringRepresentation, EntityType type, Object... parameters)
+    {
+        XWikiContext xcontext = this.xcontextProvider.get();
+        try {
+            ExtendedURL extendedURL =
+                new ExtendedURL(new URL(urlStringRepresentation), xcontext.getRequest().getContextPath());
+            ResourceType resourceType = this.typeResolver.resolve(extendedURL, Collections.emptyMap());
+            ResourceReference reference =
+                this.resourceResolver.resolve(extendedURL, resourceType, Collections.emptyMap());
+            if (reference instanceof EntityResourceReference) {
+                EntityReference entityReference = ((EntityResourceReference) reference).getEntityReference();
+                return entityReference;
+            }
+        } catch (CreateResourceReferenceException | UnsupportedResourceReferenceException | CreateResourceTypeException
+            | MalformedURLException e) {
+            logger.debug("Failed to extract an EntityReference from [{}].".concat(urlStringRepresentation), e);
```

Comment:
Why ``.concat``? I think you simply need to pass ``urlStringRepresentation`` as a parameter to ``.debug``.

Reply:
Also maybe better to use a warning there? In general you want to have those messages in the console.